### PR TITLE
Additionally Submit 674 in SubmitCentralForm686cJob

### PR DESCRIPTION
**Pull request by tfink419**
## Summary
- Updates SubmitCentralForm686cJob to check for which of 674, 686c or both forms are being uploaded.
- Rename `process_record` to `process_pdf` and move pdf creation to before method since how the to_pdf was called had to be different for each form type

## Related issue(s)
- [Original Ticket](https://app.zenhub.com/workspaces/benefits-dependents-experience-6459139e7d64bf0015f6af13/issues/gh/department-of-veterans-affairs/va.gov-team/64538)
- [Original PR](https://github.com/department-of-veterans-affairs/vets-api/pull/13775)

## Testing done
- So far only specs have been run, this will be tested in staging since it is behind a flipper

## What areas of the site does it impact?
This is related to background reprocessing of forms and does not effect the site

## Acceptance criteria

- [x]  I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x]  No error nor warning in the console.
- [ ]  ~~Events are being sent to the appropriate logging solution~~
- [ ]  ~~Documentation has been updated (link to documentation)~~
- [x]  No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [x]  Feature/bug has a monitor built into Datadog or Grafana (if applicable)
- [ ]  ~~If app impacted requires authentication, did you login to a local build and verify all authenticated routes work as expected~~
- [ ]  ~~I added a screenshot of the developed feature~~

### CODE REVIEW

File: app/models/saved_claim/dependency_claim.rb [hunk 1]
```diff
@@ -35,10 +35,7 @@ class SavedClaim::DependencyClaim < CentralMailClaim
   validate :address_exists
 
   def upload_pdf(form_id, doc_type: '148')
-    self.form_id = form_id
-
-    form_path = PdfFill::Filler.fill_form(self)
-    upload_to_vbms(path: form_path, doc_type:)
+    upload_to_vbms(path: to_pdf(form_id:), doc_type:)
```
Comment (tblackwe) [inline app/models/saved_claim/dependency_claim.rb:38]: I approve of this change so much. ty for making it more logically named.

File: app/sidekiq/central_mail/submit_central_form686c_job.rb [hunk 2]
```diff
@@ -28,19 +30,19 @@ def perform(saved_claim_id, encrypted_vet_info, encrypted_user_struct)
       Rails.logger.info('CentralMail::SubmitCentralForm686cJob running!',
                         { user_uuid: user_struct['uuid'], saved_claim_id:, icn: user_struct['icn'] })
       @claim = SavedClaim::DependencyClaim.find(saved_claim_id)
-      @claim.add_veteran_info(vet_info)
+      claim.add_veteran_info(vet_info)
```
Comment (tblackwe) [inline app/sidekiq/central_mail/submit_central_form686c_job.rb:33]: what necessitated changing this from an instance var to a local one? (seeking to understand)

**Review by tblackwe — APPROVED** [review]

File: app/sidekiq/central_mail/submit_central_form686c_job.rb (hunk 2 above)
Comment (tfink419) [reply app/sidekiq/central_mail/submit_central_form686c_job.rb:33]: Nothing necessitated it. I just have been leaning towards this style lately. I believe one thing that is nice about using `attr_reader` is that it makes it easier to track where the var has been/can be assigned

**Review by tblackwe — APPROVED** [review]

**Review by tblackwe — APPROVED** [review]

**Review by rmtolmach — APPROVED** [review]
This LGTM.

Note: because of the PACT Act deadline at EOD on the 30th, we're trying to limit merges that may impact veterans applying for benefits in the next couple days. If you're confident this won't impact that, you can go ahead and merge. However, hold off until Monday if you anticipate any impact.

**Comment by tblackwe** [comment]
TY, this will be off, behind a flipper.

**Pull request merged**